compiler/src/iree/compiler/Codegen/Common/OptimizeTensorInsertExtractSlices.cpp

```diff
@@ -84,63 +84,73 @@ hoistLoopInvariantSubsetAtIterArg(RewriterBase &rewriter,
     return loopLike;
   }
 
-  // Get all subset extraction uses of this iter_arg and try to hoist them
-  // out of the loop.
-  for (Operation *op : loopLike.getRegionIterArgs()[idx].getUsers()) {
-    auto extraction = dyn_cast<SubsetExtractionOpInterface>(op);
-    if (!extraction) {
-      continue;
-    }
+  bool changed = true;
+  while (changed) {
```

Comment:
Could we bound this loop? I think the tript count shouldn't be more than the number of users? I want to make sure this doesn't accidentally become an infinite loop.

Reply:
+1 to this. Lets add num_iters bounded by 10 or something that should be enough.